lms/static/scripts/frontend_apps/components/SubmitGradeForm.js

```diff
@@ -42,54 +82,64 @@ export default function SubmitGradeForm({ disabled = false, student }) {
     const validationError = validateGrade(value);
 
     if (validationError) {
-      setGradeErrorMessage(validationError);
-      setShowError(true);
+      setValidationMessageMessage(validationError);
+      setValidationError(true);
     } else {
+      setGradeSaving(true);
       try {
-        setRequestStatus('fetching');
-        // Divide value by 10 because the lms service layer expects grade to be a value between [0-1] but
-        // we treat this on the UI to be between [0-10].
-        await submitGrade({ student, grade: value / 10, authToken });
-        setRequestStatus('done');
+        await submitGrade({
+          student,
+          grade: value / GRADE_MULTIPLIER,
+          authToken,
+        });
+        setGradeSaved(true);
       } catch (e) {
-        setRequestStatus('error');
-        setNetworkError(e.errorMessage ? e.errorMessage : 'Unknown error');
+        setSubmitGradeError(e.errorMessage ? e.errorMessage : 'Unknown error');
       }
+      setGradeSaving(false);
     }
   };
 
   /**
    * If any input is detected, close the ValidationMessage.
    */
   const handleKeyDown = () => {
-    setShowError(false);
+    setValidationError(false);
   };
 
   return (
     <form className="SubmitGradeForm" autoComplete="off">
       <ValidationMessage
-        message={gradeErrorMessage}
-        open={showError}
+        message={validationMessage}
+        open={showValidationError}
         onClose={() => {
           // Sync up the state when the ValidationMessage is closed
-          setShowError(false);
+          setValidationError(false);
         }}
       />
       <label className="SubmitGradeForm__label" htmlFor="lms-grade">
         Grade (Out of 10)
       </label>
-      <input
-        className={classNames('SubmitGradeForm__grade', {
-          'SubmitGradeForm__grade--saved': requestStatus === 'done',
-        })}
-        disabled={disabled}
-        id="lms-grade"
-        ref={refInput}
-        onKeyDown={handleKeyDown}
-        type="input"
-        defaultValue={''} // TODO: stub until we pull the grade from the service
-        key={student.LISResultSourcedId}
-      />
+      <span className="SubmitGradeForm__grade-wrapper">
+        <input
+          className={classNames('SubmitGradeForm__grade', {
+            'SubmitGradeForm__grade--saved': gradeSaved,
```

Comment:
We had a convention in h, which I'm afraid is probably not followed everywhere, which was to use `--<modifier>` names for "static" states and "is-<state>" classes (ie. `<input class="SubmitGradeForm__grade is-saved">` for dynamic states). See https://github.com/hypothesis/frontend-toolkit/blob/master/docs/css-style-guide.md#naming (the part about not using certain classes in code is obviously not relevant for Preact/React components).